pkg/discovery/worker/compliance/affinity_processor.go

```diff
@@ -67,13 +87,14 @@ func (am *AffinityProcessor) processAffinityPerPod(pod *api.Pod, podsNodesMap ma
 		return
 	}
 
-	hostNode := podsNodesMap[pod]
-	for _, node := range am.nodes {
-		if matchesNodeAffinity(pod, node) && matchesPvNodeAffinity(pvNodeSelectorTerms, node) {
-			am.addAffinityAccessCommodities(pod, node, hostNode, nodeAffinityAccessCommoditiesSold, nodeAffinityAccessCommoditiesBought)
-		}
-		if interPodAffinityMatches(pod, node, podsNodesMap) {
-			am.addAffinityAccessCommodities(pod, node, hostNode, podAffinityCommodityDTOsSold, podAffinityCommodityDTOsBought)
+	if hostNode, ok := am.affinityPodNodesCache.Load(pod); ok {
```

Comment:
There is conditional logic based on hostNode found or not in `addAffinityAccessCommodities()`